code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,658 @@
+                                          the port has to be exposed on a specific network other then default.
+                    protocol:
+                      type: string
+                      enum:
+                        - TCP
+                        - UDP
+                        - HTTP/HTTPS
+                      description: Defines the IP transport communication protocol i.e., TCP, UDP or HTTP
+                    port:
+                      type: integer
+                      format: int32
+                      minimum: 1
+                      maximum: 65535
+                      description: Port number exposed by the component. OP may generate a dynamic port towards the UCs 
+                                          corresponding to this internal port and forward the client traffic from dynamic port to containerPort.
+                    interfaceName:
+                      type: string
+                      pattern: ^[a-z][a-z0-9]{3}$
+                      description:  Interface Name. Required only if application has to be attatched to a network other then default.
+                    visibilityType:
+                      description: Defines whether the interface is exposed to outer world or not i.e., external, or internal.
+                                          If this is set to "external", then it is  exposed to external applications otherwise it is exposed 
+                                          internally to edge application components within edge cloud. When exposed to external world, 
+                                          an external dynamic port is assigned for UC traffic and mapped to the internal containerPort
+                      type: string
+                      enum:
+                        - VISIBILITY_EXTERNAL
+                        - VISIBILITY_INTERNAL
+                minItems: 1                
+              numOfInstances:
+                type: integer
+                format: int32
+                description: Number of component instances to be launched. 
+              restartPolicy:
+                type: string
+                enum:
+                  - RESTART_POLICY_ALWAYS
+                  - RESTART_POLICY_NEVER
+                description: How the platform shall handle component failure                        
+      required:
+        - name
+        - virtType
+        - fileName
+        - repository
+        - componentSpec
+    CpuArchType:
+          type: string
+          enum:
+            - x86_64
+            - arm_64
+          description: CPU Instruction Set Architecture (ISA) E.g., Intel, Arm etc. 
+    ProvisionedApp:
+      description: Information about the provisioned application
+      type: object
+      properties:
+        appInstanceId:
+          $ref: '#/components/schemas/AppInstanceId'
+        edgeResourceName:
+          $ref: '#/components/schemas/EdgeResourceName'
+    EdgeResourceName:
```

Comment:
If EdgeResourceName refer to same parameter as used in "Simple Edge Discovery" request then we need to assess how to make a single definition for such common available across multiple APIs.

Reply:
The idea is to incorporate simple edge discovery in this set of request as it is part of the LCM of Edge Cloud. We yet belive that refering to `edgeResourceName` is more aproppiate than the existing one in the simple edge discovery. As the `edgeResource` could be refering not only to the name of the resource but also to the resources existing on a specific node. This is because in the simple edge discovery API it is not being refered the contain of the resources but only the resources itselves.